src/idpi/operators/pot_vortic.py

```diff
@@ -0,0 +1,143 @@
+    QW_load: xr.DataArray | None = None,
+    diff_type: str | None = None,
+) -> xr.DataArray:
+    """Compute the potential vorticity.
+
+    The potential vorticity is computed with the following formula:
+
+    .. math::
+        v_p = \frac{1}{\rho} * \frac{\partial \Theta}{\partial \z} * (c_v + 2 \Omega)
+
+    where :math:`\rho` is the total air density, :math:`\frac{\partial \Theta}{\partial \z}` is the gradient of the potential temperature,
+    :math:`c_v` is the curl of the wind in y direction and :math`\Omega` is the coriolis term.
+
+    The effect of water loading is ignored if ``QW_load`` is not specified.
+
+    The type of the numerical derivatives can be specified by setting ``diff_type`` to either 'left', 'right' or 'center' (default).
+    For 'center', both neighboring values will be used.
+    With 'left', only the left neighbors will be used and the right neighbor will be assumed to be equal to the current value.
+    With 'right', it's the other way around.
+    The left neighbor's index is one lower than the current index, while the right neighbor's index is one higher.
+
+    All input arrays must have the same coordinates and sizes.
+    The output will be the same orthotope as the inputs, where the boundary was removed.
+    The boundaries of some inputs are used for computing the numerical derivatives and hence, they cannot be included in the result.
+
+    Args:
+        U (xr.DataArray): Wind in x direction
+        V (xr.DataArray): Wind in y direction
+        W (xr.DataArray): Wind in z direction
+        P (xr.DataArray): Pressure
+        T (xr.DataArray): Temperature
+        HHL (xr.DataArray): Height of half-layers
+        QV (xr.DataArray): Specific humidity [kg/kg]
+        QC (xr.DataArray): Specific cloud water content [kg/kg]
+        QI (xr.DataArray): Specific cloud ice content [kg/kg].
+        QW_load (xr.DataArray | None, optional): Specific precipitable components content [kg/kg]. Defaults to None.
+        diff_type (_type_, optional): The type of differentiation. Defaults to None.
+
+    Returns:
+        xr.DataArray: The potential vorticity
+    """
+    stpt = partial(stpt_, diff_type=diff_type) if diff_type else stpt_
+
+    # prepare parameters
+    dlon = U.attrs["GRIB_iDirectionIncrementInDegrees"]
+    dlat = U.attrs["GRIB_jDirectionIncrementInDegrees"]
+    deg2rad = np.pi / 180
+    inv_dlon = 1 / (dlon * deg2rad)
+    inv_dlat = 1 / (dlat * deg2rad)
+
+    # target coordinates
+    # the output array is missing the border, so we remove the border from the coordinates as well
+    mask = {"x": slice(1, -1), "y": slice(1, -1)}
+    lat = P["latitude"][mask]
+    lon = P["longitude"][mask]
+
+    # FD (finite differences) weights
+    wi = 0.5 * inv_dlon
+    wj = 0.5 * inv_dlat
+    wk = 0.5 if diff_type == "center" else 1.0
+
+    # remove coordinates
+    U, V, W, P, T, HHL, QV, QC, QI = map(
+        lambda a: a.drop_indexes("z"), (U, V, W, P, T, HHL, QV, QC, QI)
+    )
```

Comment:
I'm not the biggest fan of this but at least it doesn't mutate the input variables. I still need to confirm that this works.